File: CSharp/SpaceVIL/SpaceVIL/Items/SplitHolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;
using SpaceVIL.Core;
using SpaceVIL.Decorations;

namespace SpaceVIL
{
    public class SplitHolder : Prototype, IDraggable
    {
        private static int count = 0;
        private Orientation _orientation;
        private int _spacerSize = 6;

        /// <summary>
        /// Constructs a SplitHolder with orientation (HORIZONTAL or VERTICAL)
        /// </summary>
        public SplitHolder(Orientation orientation)
        {
            _orientation = orientation;
            SetItemName("SplitHolder_" + count);
            count++;
            IsFocusable = false;
            MakeHolderShape();
        }

        /// <summary>
        /// SplitHolder size (height for the HORIZONTAL orientation,
        /// width for the VERTICAL orientation)
        /// </summary>
        public void SetSpacerSize(int spSize)
        {
            if (_spacerSize != spSize)
            {
                _spacerSize = spSize;
                MakeHolderShape();
            }
        }
        public int GetHolderSize()
        {
            return _spacerSize;
        }

        private void MakeHolderShape()
        {
            switch (_orientation)
            {
                case Orientation.Vertical:
                    SetSizePolicy(SizePolicy.Fixed, SizePolicy.Expand);
                    SetWidth(_spacerSize);
                    SetMinWidth(_spacerSize);
                    break;

                case Orientation.Horizontal:
                    SetSizePolicy(SizePolicy.Expand, SizePolicy.Fixed);
                    SetHeight(_spacerSize);
                    SetMinHeight(_spacerSize);
                    break;
            }
        }

        /// <returns> Orientation of the SplitHolder (Horizontal or Vertical) </returns>
        public Orientation GetOrientation()
        {
            return _orientation;
        }

        /// <summary>
        /// Set style of the SplitHolder
        /// </summary>
        public override void SetStyle(Style style)
        {
            if (style == null)
                return;
            SetBackground(style.Background);
        }
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/Items/SplitHolder.java
package com.spvessel.Items;

import com.spvessel.Flags.*;
import com.spvessel.Cores.InterfaceDraggable;
import com.spvessel.Decorations.Style;
import com.spvessel.Flags.Orientation;

public class SplitHolder extends VisualItem implements InterfaceDraggable {
    private static int count = 0;
    private Orientation _orientation;

    public Orientation getOrientation() {
        return _orientation;
    }

    public SplitHolder(Orientation orientation) {
        _orientation = orientation;
        setItemName("SplitHolder_" + count);
        count++;
        makeHolderShape();
    }

    private int _spacerSize = 6;

    public void setSpacerSize(int spSize) {
        if (_spacerSize != spSize) {
            _spacerSize = spSize;
            makeHolderShape();
        }
    }

    public int getHolderSize() {
        return _spacerSize;
    }

    private void makeHolderShape() {
        switch (_orientation) {
        case VERTICAL:
            setSizePolicy(SizePolicy.FIXED, SizePolicy.EXPAND);
            setWidth(_spacerSize);
            setMinWidth(_spacerSize);
            break;

        case HORIZONTAL:
            setSizePolicy(SizePolicy.EXPAND, SizePolicy.FIXED);
            setHeight(_spacerSize);
            setMinHeight(_spacerSize);
            break;
        }
    }

    @Override
    public void setStyle(Style style) {
        if (style == null)
            return;
        setBackground(style.background);
    }
}